### PR TITLE
Improve cpp wrapper

### PR DESCRIPTION
**Pull request by MathemanFlo**
I added two missing functions to the cplusplus wrapper.

Question: Besides `inplace` which I use for pixel modifications afterwards the c++ wrapper is missing methods for functions like `vips_draw_circle1` (with focus on the 1 versions). Would it be okay if I implemented overloads for circle, ... similar to `ifthenelse`. For example 
```
void draw_circle( ... )
{
  draw_circle( to_vector( ink), ... ); 
}
```

### CODE REVIEW

**Comment by jcupitt** [comment]
Hi @MathemanFlo,

Thanks, this looks useful.

Sure, some extra overloads for `draw_circle` etc. sound good.

**Comment by kleisauke** [comment]
FWIW, CIFuzz failure was supposed to be fixed with PR https://github.com/google/oss-fuzz/pull/9575, I'm not sure why it doesn't test against the latest OSS-Fuzz master branch.

macOS CI failure will be fixed with PR https://github.com/libvips/libvips/pull/3327.

**Comment by MathemanFlo** [comment]
- Now the C++ wrapper for the drawing functions is on par with the underlying C API
- I also implemented a void* buffer version  (instead of a `VipsBlob`) for `thumbnail_buffer`. It is now identical to the C API and easier to use.

File: cplusplus/VImage.cpp [hunk 1]
```diff
@@ -777,6 +777,28 @@ VImage::write_to_target( const char *suffix, VTarget target,
...
+	{
+		VImage thumbnail = thumbnail_buffer( blob, width, options );
+		vips_area_unref( VIPS_AREA( blob ) );
+		return( thumbnail );
+	}
+	catch ( VError &ex )
+	{
+		vips_area_unref( VIPS_AREA( blob ) );
+		throw; // rethrow exception
+	}
```
Comment (kleisauke) [inline cplusplus/VImage.cpp:799]: I think this try-catch can be avoided like this:
```suggestion
	VImage out;

	/* We don't take a copy of the data or free it.
	 */
	blob = vips_blob_new( NULL, buf, len );
	options = (options ? options : VImage::option())->
		set( "buffer", blob )->
		set( "width", width )->
		set( "out", &out );
	vips_area_unref( VIPS_AREA( blob ) );

	call( "thumbnail_buffer", options );

	return( out );
```

See for example the implementation of `VImage::new_from_buffer`:
https://github.com/libvips/libvips/blob/80f1d89365ecd69b268e7ffd22fed8b0d4f5fa62/cplusplus/VImage.cpp#L596-L606

**Comment by MathemanFlo** [comment]
@kleisauke Thank you for your suggestion. I thought the blob must not be destroyed before `call`.

**Review by kleisauke — APPROVED** [review]
Thank you!

**Comment by MathemanFlo** [comment]
Thank you for letting me participate

**Pull request merged**